### PR TITLE
Unable to use custom handlers for HTTP OPTIONS method in subresources

### LINKED ISSUE #45173: Unable to use custom handlers for HTTP OPTIONS method in subresources
### Describe the bug

[**Breaking change in 3.15.2 minor release**]

After merging https://github.com/quarkusio/quarkus/pull/43440, we (Keycloak) are not able to use our custom logic for handling the HTTP OPTIONS method in Quarkus REST subresources. When there's an HTTP request with the OPTIONS method made for a subresource that does not explicitly contain a handler annotated with `@OPTIONS`, [default values](https://github.com/quarkusio/quarkus/blob/main/independent-projects/resteasy-reactive/server/runtime/src/main/java/org/jboss/resteasy/reactive/server/handlers/ResourceLocatorHandler.java#L69) are immediately returned. However, when some specific handlers are nested deeper in subresources, they are not taken into account. 

Previously, all subresources were traversed and checked whether they contain the handler for `@HEAD`, and `@OPTIONS` HTTP methods.

### Expected behavior

Use custom nested handlers for `@OPTIONS` and `@HEAD` HTTP methods. 

### Actual behavior

NOT POSSIBLE to use custom nested handlers for `@OPTIONS` and `@HEAD` HTTP methods. 

### How to Reproduce?

1. Have custom handlers for `OPTIONS` and `HEAD` methods in sub-resources
2. Make an OPTIONS/HEAD HTTP request for them, or leverage preflight logic
3. See the custom handler is not executed, but some default response is returned

### Output of `uname -a` or `ver`

Linux mabartos 6.11.9-100.fc39.x86_64 #1 SMP PREEMPT_DYNAMIC Sun Nov 17 18:52:19 UTC 2024 x86_64 GNU/Linux

### Output of `java -version`

openjdk 21.0.3 2024-04-16 LTS OpenJDK Runtime Environment Temurin-21.0.3+9 (build 21.0.3+9-LTS) OpenJDK 64-Bit Server VM Temurin-21.0.3+9 (build 21.0.3+9-LTS, mixed mode, sharing)

### Quarkus version or git rev

3.15.2

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Maven 3.9.8

### Additional information

I'll provide a PR for this. 

_Maybe it would be good to go through the same case in the `ClassRoutingHandler` containing also the default mechanisms for OPTIONS/HEAD. But it's out of scope of this issue._

### PR DESCRIPTION
**Pull request by mabartos**
- Fixes #45173

More details are in the attached issue.

### CODE REVIEW

File: independent-projects/resteasy-reactive/server/runtime/src/main/java/org/jboss/resteasy/reactive/server/handlers/ResourceLocatorHandler.java [hunk 1]
```diff
@@ -63,29 +63,30 @@ public void onComplete(Throwable throwable) {
...
+                            if (method == null) {
+                                continue;
+                            }
+                            allowedMethods.add(method);
+                        }
+                        allowedMethods.add(HttpMethod.OPTIONS);
+                        allowedMethods.add(HttpMethod.HEAD);
+                        requestContext.abortWith(Response.ok().allow(allowedMethods).build());
+                        return;
+                }
```
Comment (geoand) [inline independent-projects/resteasy-reactive/server/runtime/src/main/java/org/jboss/resteasy/reactive/server/handlers/ResourceLocatorHandler.java:89]: Can we convert this back to an if else please?

Comment (mabartos) [reply independent-projects/resteasy-reactive/server/runtime/src/main/java/org/jboss/resteasy/reactive/server/handlers/ResourceLocatorHandler.java]: @geoand done

**Review by geoand — APPROVED** [review]
Thanks a lot!

**Comment by mabartos** [comment]
@geoand Ahh, I had to run the formatter to pass validations.

**Review by cescoffier — APPROVED** [review]

**Comment by FroMage** [comment]
You need to rebase this on the latest `main` if you want quickstarts to compile. Otherwise ignore the issue of quickstarts if the only problem is webauthn because this is working on the latest `main` tips on both repos.

**Comment by geoand** [comment]
Thanks for checking @FroMage!

I rebased the PR, so let's see what happens

**Pull request merged**